base_classes/NXdata.nxdl.xml

```diff
@@ -85,124 +85,170 @@
 		.. index:: axes (attribute)
 		.. index:: coordinates
 
-		The :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields contain the axis coordinates associated with the data values.
-		The names of all :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields are listed in the
-		:ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute.
+		The :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields contain the axis coordinates associated with the signal values.
 
-		`Rank`
+		`Default axes`
 
-		:ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields are typically one-dimensional arrays, which annotate one of the dimensions.
+		The :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute provides the names of the :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;`
+		fields to be used as the default axis for each dimension of the :ref:`data &lt;/NXdata/DATA-field&gt;` fields.
+		As a result the length of the :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute must be equal to the rank of the :ref:`data &lt;/NXdata/DATA-field&gt;`
+		fields. When a particular dimension has no default axis, the string “.” is used in that position.
 
-		An example of this would be
+		`Spanned dimensions`
+
+		The :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attributes describe the
+		:ref:`data &lt;/NXdata/DATA-field&gt;` dimensions spanned by the corresponding :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` fields.
+
+		When :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` is missing for a given :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field,
+		the position(s) of the :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field name in the :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute
+		is(are) used. The shape of an :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field must be equal to the shape of the
+		:ref:`data &lt;/NXdata/DATA-field&gt;` dimensions it spans.
+
+		Note that an :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field can have more than one dimension and can therefore span
+		more than one :ref:`data &lt;/NXdata/DATA-field&gt;` dimension. Conversely, one :ref:`data &lt;/NXdata/DATA-field&gt;` dimension
+		can be spanned by more than one :ref:`AXISNAME &lt;/NXdata/AXISNAME-field&gt;` field. As stated before, the default
+		axis name (if any) can be found in the :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute.
+
+		`Available axes`
+
+		With the current definition of the :ref:`axes &lt;/NXdata@axes-attribute&gt;` and :ref:`AXISNAME_indices &lt;/NXdata@AXISNAME_indices-attribute&gt;` attributes
+		it is not straightforward to discover all available axes. All strings in the :ref:`axes &lt;/NXdata@axes-attribute&gt;` attribute
```

Comment:
"~it is not straightforward to discover~does not provide a list of all available axes"

Reply:
I wanted to be clear that we are well aware this is a horrible and non-deterministic way of discovering axes. "not straightforward" is already a euphemism imo. But ok I'll play nice.

Reply:
Your correction did not match the sentence. I did

~it is not straightforward to discover all available axes~ a list of all available axes is not provided directly